File: native/lib/utils/common/upload_image.dart
import 'dart:io';
import 'package:firebase_storage/firebase_storage.dart';

Future<String> uploadImage(File image, String userId) async {
  final storage = FirebaseStorage.instance;
  final ref = storage.ref().child('images/users/$userId/thumbnail');
  final imageSnap = await ref.putFile(image);
  final imageURL = await imageSnap.ref.getDownloadURL();
  return imageURL;
}

File: web/plugins/uploadImage.ts
import firebase from '../plugins/firebase'
import { IUser } from '../types/user'

const uploadImage = async (image: string, userId: IUser['id']): Promise<string> => {
  const storage = firebase.storage()
  const ref = storage.ref().child(`images/users/${userId}/thumbnail`)
  const imageSnap = await ref.putString(image.split(',')[1], 'base64')
  const imageURL = await imageSnap.ref.getDownloadURL()
  return imageURL
}

export default uploadImage
